ballerina/tests/test_services.bal

```diff
@@ -1142,3 +1142,61 @@ service /nullable_inputs on basicListener {
         return account.number;
     }
 }
+
+public string[] namesArray = ["Walter", "Skyler"];
+
+service /subscriptions on basicListener {
+    isolated resource function get name() returns string {
+        return "Walter White";
+    }
+
+    resource function subscribe name() returns stream<string, error?> {
+        return namesArray.toStream();
+    }
+
+    isolated resource function subscribe messages() returns stream<int, error?> {
+        int[] intArray = [1, 2, 3, 4, 5];
+        return intArray.toStream();
+    }
+
+    isolated resource function subscribe stringMessages() returns stream<string?, error?> {
+        string?[] stringArray = [(), "1", "2", "3", "4", "5"];
+        return stringArray.toStream();
+    }
+
+    isolated resource function subscribe books() returns stream<Book, error?> {
+        Book[] books = [
+            {name: "Crime and Punishment", author: "Fyodor Dostoevsky"},
+            {name: "A Game of Thrones", author: "George R.R. Martin"}
+        ];
+        return books.toStream();
+    }
+
+    isolated resource function subscribe students() returns stream<StudentService, error?> {
+        StudentService[] students = [new StudentService(1, "Eren Yeager"), new StudentService(2, "Mikasa Ackerman")];
+        return students.toStream();
+    }
+
+    isolated resource function subscribe filterValues(int value) returns stream<int, error?> {
+        int[] intArray = [1, 2, 3, 4, 5];
+        int[] filteredArray = [];
+        foreach int i in intArray {
+            if i < value {
+                filteredArray.push(i);
+            }
+        }
+        return filteredArray.toStream();
+    }
+
+    isolated resource function subscribe values() returns stream<int> {
+        int[] array = [];
+        foreach int i in 1 ..< 2 {
+            int _ = array.remove(0);
+        }
```

Comment:
What's the purpose of using `foreach` loop?

Reply:
To return an unexpected error here. So the `notifyFailure()` method in the `SubscriptionCallback` class will be invoked.

Reply:
I will remove it. It is not necessary